boardswarm-cli/src/ui.rs

```diff
@@ -210,36 +206,64 @@ pub async fn run_ui(
                 async move {
                     match i {
                         Input::PowerOn => {
-                            device.change_mode("on").await.unwrap();
+                            // Perform power on action
+                            if let Err(err) = device.change_mode("on").await {
+                                let err_msg: String =
+                                    format!("Change mode 'on' action failed: {}\r\n", err.code());
+                                input_tx.send(Input::UIMessage(err_msg)).await.unwrap();
+                            }
                             None
                         }
                         Input::PowerOff => {
-                            device.change_mode("off").await.unwrap();
+                            // Perform power off action
+                            if let Err(err) = device.change_mode("off").await {
+                                let err_msg: String =
+                                    format!("Change mode 'off' action failed: {}\r\n", err.code());
+                                let _ = input_tx.send(Input::UIMessage(err_msg)).await;
+                            };
                             None
                         }
                         Input::PowerReset => {
-                            device.change_mode("off").await.unwrap();
-                            device.change_mode("on").await.unwrap();
+                            // Perform power off action
+                            if let Err(err) = device.change_mode("off").await {
+                                let err_msg: String =
+                                    format!("Change mode 'off' action failed: {}\r\n", err.code());
+                                let _ = input_tx.send(Input::UIMessage(err_msg)).await;
+                            };
+                            // Perform power on action
+                            if let Err(err) = device.change_mode("on").await {
+                                let err_msg: String =
+                                    format!("Change mode 'on' action failed: {}\r\n", err.code());
+                                let _ = input_tx.send(Input::UIMessage(err_msg)).await;
+                            }
                             None
                         }
                         Input::Up | Input::Down | Input::ScrollReset => {
-                            input_tx.send(i).await.unwrap();
+                            let _ = input_tx.send(i).await;
```

Comment:
not a blocker; but if we can send on this channels something is very very wrong ofcourse

Reply:
Indeed, makes sense to panic here, as it means something is really broken.